### DISCUSSION TITLE
saving a new nested array having properties of dynamic dimension

### DISCUSSION THREAD

**Discussion by gsaha009** [scikit-hep/awkward#3276] [General]
Hello experts,

I have a main awkward record named `events` which has several fields like `muon`, `tau`, `met` etc. Now, I want to add a new field to the `events` array e.g. `higssCand` which looks like the following:

Say, I have three entries in the `events` array, and this `higgsCand` is basically a pair of `muon` and `tau`
```py
higgsCand = [ [m,t], [m,t], [m,t] ]
# fields of higgsCand
pt : [ [45.5, 43.2], [34.3, 32.1], [87.1,54,2] ]
# more fields of the same dimension
# now is there any way to add something like the following
mT_comb: [ [41.2], [54.4], [74.6] ] # in this same higgsCand array?
```
So, if I call `events.higgsCand.pt`, it will return the `pt` array, and if I do `events.higgsCand.mT_comb`, it will return me the `mT_comb` array which has a different dimension than `pt`

Is it fundamentally possible at all?

Thank you very much :)

Gourab

**Comment by jpivarski** [comment]
Do you mean something other than the following?

```python
events["higgsCand"] = higgsCand
```

(See [`ak.Array.__setitem__`](https://awkward-array.org/doc/main/reference/generated/ak.Array.html#ak.Array.__setitem__).)
